crates/sui-framework/packages/sui-framework/sources/kiosk/kiosk.move

```diff
@@ -412,6 +415,181 @@ module sui::kiosk {
         coin::take(&mut self.profits, amount, ctx)
     }
 
+    // === Kiosk Extensions API ===
+
+    /// Extension is trying to access a permissioned action while being disabled.
+    const EExtensionDisabled: u64 = 13;
+    /// Extension is trying to access a permissioned action while not having
+    /// the required permission.
+    const EExtensionNotAllowed: u64 = 14;
+    /// Extension is trying to access a permissioned action while not being
+    /// authorized to use the type.
+    const EExtensionNotAllowedForType: u64 = 15;
+
+    /// The Extension struct contains the data used by the extension and the
+    /// configuration for this extension. Stored under the `ExtensionKey`
+    /// dynamic field.
+    struct Extension has store {
+        /// Storage for the extension, an isolated Bag. By putting the extension
+        /// into a single dynamic field, we reduce the amount of fields on the
+        /// top level (eg items / listings) while giving extension developers
+        /// the ability to store any data they want.
+        storage: Bag,
+        /// Bitmask of permissions that the extension has (can be revoked any
+        /// moment). It's all or nothing policy - either the extension has the
+        /// required permissions or no permissions at all.
+        permissions: u32,
+        /// List of types that the extension is allowed to use. It only affects
+        /// permissioned actions (eg `place`, `lock`). The Kiosk owner can limit
+        /// an extensions functionality to a set of specified types.
+        types: vector<TypeName>,
```

Comment:
I would anticipate that folks will want to use an allowlist and/or a denylist in specifying which types an extension can access (just about any system I've seen for one ends up wanting the other), so we should probably include fields for both.

Reply:
Agree that `allowed_types` fits it better. However, Extension is never available to anyone and intentionally abstracted away so that we can change the format later. Having said that, working with a vector without an Option simplifies things a lot (remember how many issues developers face with Option in a PTB).

Reply:
>I would anticipate that folks will want to use an allowlist and/or a denylist in specifying which types an extension can access (just about any system I've seen for one ends up wanting the other), so we should probably include fields for both.

In this regard I don't think anyone would ever use denylist. The goal of this setting is to protect the Kiosk from being used on types the Kiosk Owner doesn't want to see. Eg I only want Capy Mixer extension to add a new Capy while I don't want it to add any other object. 

One can always create a new type for scammy-spammy object and having a denylist for that purpose seems like a waste of time.